lib/tapioca/compilers/dsl/active_record_scope.rb

```diff
@@ -0,0 +1,92 @@
+# typed: true
+# frozen_string_literal: true
+
+require "parlour"
+
+begin
+  require "active_record"
+rescue LoadError
+  return
+end
+
+module Tapioca
+  module Compilers
+    module Dsl
+      # `Tapioca::Compilers::Dsl::ActiveRecordScope` decorates RBI files for subclasses of
+      # `ActiveRecord::Base` which declare `scope` fields
+      # (see https://api.rubyonrails.org).
+      #
+      # For example, with the following `ActiveRecord::Base` subclass:
+      #
+      # ~~~rb
+      # class Post < ApplicationRecord
+      # end
+      # ~~~
+      #
+      # this generator will produce the RBI file `post.rbi` with the following content:
+      #
+      # ~~~rbi
+      # # post.rbi
+      # # typed: true
+      # class Post
+      # end
+      # ~~~
+      class ActiveRecordScope < Base
+        extend T::Sig
+
+        sig do
+          override.params(
+            root: Parlour::RbiGenerator::Namespace,
+            constant: T.class_of(::ActiveRecord::Base)
+          ).void
+        end
+        def decorate(root, constant)
+          scopes = constant.send(:generated_relation_methods).instance_methods(false)
+          return if scopes.blank?
+          module_name = "#{constant}::GeneratedRelationMethods"
+          root.create_module(module_name) do |mod|
+            generate_instance_methods(constant, mod)
```

Comment:
```suggestion
            generate_instance_methods(constant, mod, scope_methods)
```

Reply:
I dont have any such variable as `scope_methods`, and Why do we need to pass `scope_methods` ?

Reply:
I added the lookup in decorate and added `scope_method`  and deleted `constant` since there wont be no need for it.